src/RatingAPILayer.js

```diff
@@ -14,10 +14,11 @@ class RatingAPILayer extends React.PureComponent {
     this.handleHover = this.handleHover.bind(this);
   }
 
-  UNSAFE_componentWillReceiveProps(nextProps) {
-    this.setState({
-      value: nextProps.initialRating
-    });
+  static getDerivedStateFromProps(props, prevState) {
+    const { initialRating } = this.props;
```

Comment:
Ups. `this.props` -> `props`
So used to write always together...

Reply:
:man_facepalming: good catch.